### PR TITLE
Docs: update developer guide and other minor tweaks

### PR DESCRIPTION
**Pull request by bnjmnt4n**
- Updates "UI component" and "Common classes" sections of Developer Guide, including the UML diagram.
- Various minor tweaks to the docs.

### CODE REVIEW

File: docs/DevOps.md [hunk 1]
```diff
@@ -44,7 +44,7 @@ As part of CI, this project uses Codecov to generate coverage reports. Here are
 
 1. Sign up with Codecov using your GitHub account [here](https://codecov.io/signup).
 1. Once you are inside Codecov web app, add your fork to CodeCov.
-1. Get the Markdown code for the Codecov badge provided in `Settings > Badges` and update the `docs/index.md` of your repo with it so that the badge [![codecov](https://codecov.io/gh/se-edu/addressbook-level3/branch/master/graph/badge.svg)](https://codecov.io/gh/se-edu/addressbook-level3) in that page reflects the coverage of your project.
+1. Get the Markdown code for the Codecov badge provided in `Settings > Badges` and update the `docs/index.md` of your repo with it so that the badge [![codecov](https://codecov.io/gh/AY2122S1-CS2103T-F11-4/tp/branch/master/graph/badge.svg)](https://codecov.io/gh/AY2122S1-CS2103T-F11-4/tp) in that page reflects the coverage of your project.
```
Comment (david-eom) [inline docs/DevOps.md:47]: Nice one!

File: docs/UserGuide.md [hunk 2]
```diff
@@ -42,7 +42,7 @@ This section contains a glossary of all the commands and quick examples for refe
 
 1. Ensure you have `Java 11` or above installed on your Computer.
    If you already have `Java` installed, you might want to type in `java -version` in your command window to ensure you are running on the correct version of `Java`!
-   If you don't have `Java 11`, simply click [here](https://www.oracle.com/java/technologies/downloads/), and navigate to the "Java SE Development Kit 11.*.**" section to download the latest Java Development Kit.
+   If you don't have `Java 11`, simply click [here](https://www.oracle.com/java/technologies/downloads/), and navigate to the "Java SE Development Kit 11.x.xx" section to download the latest Java Development Kit.
```
Comment (david-eom) [inline docs/UserGuide.md:45]: Yeap in hindsight this makes more sense to non-technical users.

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -90,13 +90,13 @@ The sections below give more details of each component.
 
 ### UI component <a name="ui"></a>
 
-The **API** of this component is specified in [`Ui.java`](https://github.com/AY2122S1-CS2103T-F11-4/tp/blob/master/src/main/java/socialite/ui/Ui.java)
+The `UI` component interacts with other components of SociaLite, and uses the JavaFx UI framework to render a GUI.
```
Comment (david-eom) [inline docs/DeveloperGuide.md:93]: Minor nit: change from `JavaFx` to `JavaFX`.

**Review by david-eom — APPROVED** [review]
Hi Ben, thanks for the UG&DG update! LGTM except a small nit for the capitalisation of JavaFX. Thanks for pointing out the `PersonCard --> MainWindow` reference as well 👍

File: docs/DevOps.md (hunk 1 above)
Comment (stanley-1) [reply docs/DevOps.md:47]: Thanks for this catch @bnjmnt4n!

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -105,6 +105,8 @@ The `UI` component,
 * keeps a reference to the `Logic` component, because the `UI` relies on the `Logic` to execute commands.
 * depends on some classes in the `Model` component, as it displays `Person` object residing in the `Model`.
 
+The `PersonCard` part also has a reference to the `MainWindow` to provide improved user feedback via the main window after user interactions which do not execute a command using `Logic`.
+cc
```
Comment (stanley-1) [inline docs/DeveloperGuide.md:109]: Need to remove this 'cc'

**Review by stanley-1 — APPROVED** [review]
Thanks for the additions to the DG! Just have to delete a random 'cc' as indicated below.

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (bnjmnt4n) [reply docs/DeveloperGuide.md:109]: Whoops, not sure how that got there.

Comment (bnjmnt4n) [reply docs/DeveloperGuide.md]: Removed in https://github.com/AY2122S1-CS2103T-F11-4/tp/pull/169/commits/72e7b39a9b3b4b6d8fa75d3c44d74ddfb3f72e6c.

**Pull request merged**